### PR TITLE
RoR Capstone project 

### PR DESCRIPTION
**Pull request by Wahaj-Ali**
Hi 👋 

## HIGHLIGHTS of the project
This web app has the following features:
#### Interactions
- Sign up and login pages.
- Home page (categories page).
- Transactions page.
- For each category, the user can see the category name, icon, and total amount of all the transactions that belong to the category.
- Two forms to add new categories or new transactions.

There is a button "add transaction" at the bottom that brings the user to the page to create a new transaction, and upon saving it adds a new transaction to the transactions list and redirects the user to the home page.
When the user clicks on the "Back" button (<), the user navigates to the home page.

Similarly, the "Add Category" button creates a new category, and upon saving it adds a new category to the categories list and redirects the user to the home page.
When the user clicks on the "Back" button (<), the user navigates to the home page.

- Unit & Integraton test

Add unit and integration tests for all the most important components using RSpec.

- Deployment

Use Render to deploy the App. [Link](https://budget-app-eydq.onrender.com/)

#### Technical requirements
- Use Postgres as your database.
- Use devise for authentication.
- Add validations so that no one with bad intentions can compromise your app.
- Project is deployed and accessible online.
- The database scheme matches the given ERD diagram.

[Project presentation Video](https://www.loom.com/share/c9731690e7044d419e01962d1f4b598a) (UPDATED)

### CODE REVIEW

File: app/views/money_tracks/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<%=form_with model: [@group, @group.money_tracks.create], class:"input-form"  do |f|%>
+
+  <nav class="header">
+  <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+    <li>
+    <a href="/"><i class="fa fa-arrow-left"></i></a>
+    </li>
```
Comment (codepantha) [inline app/views/money_tracks/new.html.erb]: - [x] From the "Add Transaction" page, when a user clicks on the "Back" button (<), the user should navigate to the transactions page for that category. At the moment, the "Back" button (<) button takes us to the homepage. Please make this change 🙏

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+   <div class="category-info">
+    <p> <%=image_tag "#{group.icon}" ,class:"p-2 m-1 rounded img-thumbnail"%></p>
+    <ul class="p-2 m-1">
+      <li> <p class="d-flex cat_name"><%= link_to group.name, group_path(group), class:"p-2 m-2 text-muted h2" %></p></li>
+      <li> <p class="p-2 m-2 text-muted cat_date"> <%= group.created_at.strftime("%d-%b-%Y")%></p></li>
+    </ul>
+    </div>
+    <ul class="p-2 m-1">
+    <li><p class="p-2 m-1 h3"> $<%= group.money_tracks.sum(:amount)%></p></li>
+    <li>
```
Comment (codepantha) [inline app/views/groups/index.html.erb]: - [x] Please follow the design guidelines in terms of `font-family` and `font-weight` here and across your application.
- [x] I also noticed that the category's name is cut off even though there's more than enough room to contain it. Please try to fix it in order to improve your UI 👍 
### Design Guideline
image

### Your Design
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Wahaj-Ali 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

- [ ] According to the requirements, the video should contain a highlight about a piece of the code that you find interesting or are proud of making. Kindly re-do the video and incorporate this suggestion 👍 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/groups/index.html.erb (hunk 2 above)
Comment (Wahaj-Ali) [reply app/views/groups/index.html.erb]: Hi @codepantha, I tried to find the font on google fonts, but it is not available. So I won't be able to use the exact font family in my App.

File: app/views/money_tracks/new.html.erb (hunk 1 above)
Comment (Wahaj-Ali) [reply app/views/money_tracks/new.html.erb]: @codepantha thank you for pointing out this issue, surely making this change will make my App better.

**Comment by Wahaj-Ali** [comment]
Thank you @codepantha for your valuable suggestions, I am sure these suggestions will make it the best version of my App.

File: app/views/money_tracks/new.html.erb (hunk 1 above)
Comment (Wahaj-Ali) [reply app/views/money_tracks/new.html.erb]: @codepantha I found the font used in the design. I have updated my App as well.

File: app/views/groups/index.html.erb (hunk 2 above)
Comment (Wahaj-Ali) [reply app/views/groups/index.html.erb]: @codepantha I found the font used in the design. I have updated my App as well.

**Review by sumairq — APPROVED** [review]
### Hi @Wahaj-Ali , 



In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**